code/workspaces/web-app/src/components/settings/EditProjectDetails.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { Field, reduxForm } from 'redux-form';
+import { withStyles } from '@material-ui/core/styles';
+import { useCurrentProject } from '../../hooks/currentProjectHooks';
+import projectActions from '../../actions/projectActions';
+import syncValidate from '../projects/updateProjectFormValidator';
+import { renderTextField, renderTextArea, UpdateFormControls } from '../common/form/controls';
+import notify from '../common/notify';
+
+const styles = theme => ({
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  input: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+});
+
+export default function EditProjectDetails() {
+  const project = useCurrentProject().value;
+  const dispatch = useDispatch();
+
+  const updateProject = async (values) => {
+    try {
+      const projectKey = project.key;
+      const projectObject = { projectKey, ...values };
+      await dispatchEditProjectDetails(projectObject, dispatch);
+      notify.success('Project updated');
+    } catch (error) {
+      notify.error(`Error updating Project: ${error.message}`);
+    }
+  };
+
+  return (
+    <EditProjectReduxForm
+      project={project}
+      onSubmit={updateProject}
+    />
+  );
+}
+
+// Redux-Form Component
+
+function EditProject({ classes, project, handleSubmit }) {
+  return (
+    <form
+      className={classes.form}
+      onSubmit={handleSubmit}
+    >
+      <Field
+        name="name"
+        label="Name"
+        component={renderTextField}
+        margin="dense"
+        InputLabelProps={{ shrink: true }}
+        placeholder={project.name}
+        className={classes.input}
+      />
+      <Field
+        name="description"
+        label="Description"
+        component={renderTextArea}
+        className={classes.input}
+        margin="dense"
+        InputLabelProps={{ shrink: true }}
```

Comment:
It looks like these same props are set across all of the fields here. Might be worth grouping them into an object at the top of the function and then destructuring them into the fields.

Something like
```js
const commonProps = {
  component: renderTextArea,
  className: classes.input,
  margin: "dense",
  InputLabelProps: { shrink: true },
};

...

<Field 
  {...commonProps}
  name="name"
 ...
/>
```

Reply:
> What you have done looks really good. There are a couple of things it might be worth considering:
> 
>     1. Is there value in having a cancel button next to the apply button that just reverts the text in the description? I don't know whether that would be conforting to users.
> 
>     2. Is there value in having a confirmation dialog when apply is pressed? Thinking it might be nice as there is no undo so would be rather annoying if apply was pressed accidentally!

Both good points! Have added a cancel button and confirmation dialog.